test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs

```diff
@@ -0,0 +1,206 @@
+            
+            var siteMapData = siteMapService.GetSitemapData(requestUrl);
+
+            Assert.True(siteMapData.Equals(expectedSitemapData));
+        }
+
+        [Fact]
+        public void One_Host_And_Multiple_Sitemaps_Can_Retrieve_Correct_SiteMap()
+        {
+            var requestUrl = "https://xyz.com/en/sitemap.xml";
+            var expectedSitemapData = new SitemapData
+                {Language = "en", Host = "Sitemap.xml", SiteUrl = "https://xyz.com/"};
+
+            var hostDefinition = new HostDefinition();
+            var siteDefinition = new SiteDefinition();
+            siteDefinition.Hosts = new List<HostDefinition>
+            {
+                new HostDefinition {Name = "xyz.com"}
+            };
+
+            var siteDefinitionResolver = new Mock<ISiteDefinitionResolver>();
+            siteDefinitionResolver
+                .Setup(x => x.GetByHostname(It.IsAny<string>(), It.IsAny<bool>(), out hostDefinition))
+                .Returns(siteDefinition);
+
+            var sitemapDataList = new List<SitemapData>
+            {
+                new SitemapData { Language = "en", Host = "Sitemap.xml", SiteUrl = "https://abc.xyz.com/" },
+                expectedSitemapData
+            };
+
+            var sitemapLoader = new Mock<ISitemapLoader>();
+            sitemapLoader
+                .Setup(x => x.GetAllSitemapData())
+                .Returns(sitemapDataList);
+
+            var siteMapService = new SitemapRepository(_languageBranchRepository.Object, siteDefinitionResolver.Object, sitemapLoader.Object);
+            
+            var siteMapData = siteMapService.GetSitemapData(requestUrl);
+
+            Assert.True(siteMapData.Equals(expectedSitemapData));
+        }
+
+        [Fact]
+        public void Multiple_Host_And_Multiple_Sitemaps_Can_Retrieve_Correct_SiteMap()
+        {
+            var requestUrl = "https://abc.xyz.com/en/sitemap.xml";
+            var expectedSitemapData = new SitemapData
+                { Language = "en", Host = "Sitemap.xml", SiteUrl = "https://xyz.com/" };
+
+            var hostDefinition = new HostDefinition();
+            var siteDefinition = new SiteDefinition();
+            siteDefinition.Hosts = new List<HostDefinition>
+            {
+                new HostDefinition {Name = "xyz.com"},
+                new HostDefinition {Name = "abc.xyz.com"}
+            };
+
+            var siteDefinitionResolver = new Mock<ISiteDefinitionResolver>();
+            siteDefinitionResolver
+                .Setup(x => x.GetByHostname(It.IsAny<string>(), It.IsAny<bool>(), out hostDefinition))
+                .Returns(siteDefinition);
+            
+            var sitemapDataList = new List<SitemapData>
+            {
+                expectedSitemapData
+            };
+
+            var sitemapLoader = new Mock<ISitemapLoader>();
+            sitemapLoader
+                .Setup(x => x.GetAllSitemapData())
+                .Returns(sitemapDataList);
```

Comment:
Shouldn't this list should contain two sitemaps: one for `xyz.com` and one for `abc.xyz.com` ?

Reply:
And have to check if the order in the list matters, perhaps add two unit tests?

Reply:
Using the `InlineData` attribute to test several scenarios